File: remote/src/main/java/com/kelsos/mbrc/interactors/TrackLyricsInteractorImpl.java
package com.kelsos.mbrc.interactors;

import com.google.inject.Inject;
import com.kelsos.mbrc.repository.TrackRepository;

import java.util.List;

import rx.Observable;

public class TrackLyricsInteractorImpl implements TrackLyricsInteractor {

  @Inject private TrackRepository repository;

  @Override
  public Observable<List<String>> execute(boolean reload) {
    return repository.getTrackLyrics(reload);
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/interactors/TrackLyricsInteractorImpl.kt
package com.kelsos.mbrc.interactors

import com.kelsos.mbrc.repository.TrackRepository
import rx.Observable
import javax.inject.Inject

class TrackLyricsInteractorImpl
@Inject constructor(private val repository: TrackRepository) :
    TrackLyricsInteractor {

  override fun execute(reload: Boolean): Observable<List<String>> {
    return repository.getTrackLyrics(reload)
  }
}
